Desktop_Interface/uartstyledecoder.cpp

```diff
@@ -2,41 +2,43 @@
 #include <QDebug>
 #include <cassert>
 
-uartStyleDecoder::uartStyleDecoder(QObject *parent_in) : QObject(parent_in)
+uartStyleDecoder::uartStyleDecoder(QObject *parent_in)
+	: QObject(parent_in)
+	, parent{static_cast<isoBuffer*>(parent_in)}
```

Comment:
If parent is always an `isoBuffer`, could we just use the `isoBuffer()` constructor and get rid of the `QObject` stuff completely?